mixer/adapter/svcctrl/handler_test.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package svcctrl
+
+import (
+	"testing"
+
+	"istio.io/istio/mixer/adapter/svcctrl/template/svcctrlreport"
+	"istio.io/istio/mixer/template/apikey"
+	"istio.io/istio/mixer/template/quota"
+)
+
+func TestHandlerInterface(t *testing.T) {
+	var h interface{} = &handler{}
+	_, ok := h.(apikey.Handler)
+	if !ok {
+		t.Error(`svcctrl.handler doesn't implement apikey.Handler interface`)
+	}
+	_, ok = h.(svcctrlreport.Handler)
+	if !ok {
+		t.Error(`svcctrl.handler doesn't implement svcctrlreport.Handler interface`)
+	}
+	_, ok = h.(svcctrlreport.Handler)
+	if !ok {
```

Comment:
why are there two identical blocks here?

Reply:
that is the fault of copy-paste. Removed.